src/handlers/authenticationHandlers.js

```diff
@@ -0,0 +1,63 @@
+import axios from 'axios';
+
+import { pathObj } from './../utils/generalVariables';
+
+export const loginHandler = ({ username, password }) => {
+  // With the finalization of the database schema, more checks can be implemented (with separate error-messages)
+  if (!username || !password || typeof username !== "string" || typeof password !== "string" || password.length < 8 ) {
+    return new Error("Make sure you're passing a valid username and a password that's at least 8 characters long")
+  }
+
+  return axios.post(`${pathObj.loginPath}`, { 
+    username, 
+    password,
+  })
+    .then(res => {
+      if (res.token) {
+        checkAndStoreToken(res.data.token);
+        return res.data.user
+      } else {
+        return new Error('Oh no, there was no token returned by the database!')
+      }
+    })
+    .catch(error => {
+      return new Error(error)
```

Comment:
why did you decide to return the error inside a new error instead of just return error?

Reply:
`error` refers to the error message returned by the database in this case. Therefore, to make sure an actual error gets logged, returning a new error.